src/Std/Data/DHashMap/Lemmas.lean

```diff
@@ -964,4 +963,855 @@ theorem distinct_keys [EquivBEq α] [LawfulHashable α] :
     m.keys.Pairwise (fun a b => (a == b) = false) :=
   Raw₀.distinct_keys ⟨m.1, m.2.size_buckets_pos⟩ m.2
 
+@[simp]
+theorem insertMany_nil :
+    m.insertMany [] = m :=
+  Subtype.eq (congrArg Subtype.val (Raw₀.insertMany_nil ⟨m.1, m.2.size_buckets_pos⟩) :)
+
+@[simp]
+theorem insertMany_list_singleton {k : α} {v : β k} :
+    m.insertMany [⟨k, v⟩] = m.insert k v :=
+  Subtype.eq (congrArg Subtype.val (Raw₀.insertMany_list_singleton ⟨m.1, m.2.size_buckets_pos⟩) :)
+
+theorem insertMany_cons {l : List ((a : α) × β a)} {k : α} {v : β k} :
+    m.insertMany (⟨k, v⟩ :: l) = (m.insert k v).insertMany l :=
+  Subtype.eq (congrArg Subtype.val (Raw₀.insertMany_cons ⟨m.1, m.2.size_buckets_pos⟩) :)
+
+@[simp]
+theorem contains_insertMany_list [EquivBEq α] [LawfulHashable α]
```

Comment:
As appropriate, there should be `∈` variants of the lemmas about `contains`, see for example `mem_insert`.